news-explorer/src/components/SearchForm/SearchForm.js

```diff
@@ -1,17 +1,39 @@
 import React from "react";
-import './SearchForm.css';
+import "./SearchForm.css";
+
+export default function SearchForm({ onSearch }) {
+  const [keyword, setKeyword] = React.useState("");
+
+  function handleKeywordChange(e) {
+    // eslint-disable-next-line no-useless-escape
+    const filteredValue = e.target.value.replace(/[*|\"<>[\]{}`;&$]+/, " "); // Disallow special characters to prevent XSS
+    setKeyword(filteredValue);
+    localStorage.setItem('keyword', filteredValue) //Save keyword to local storage for retriving after refreshing the page
+  }
+
+  function handleSubmit(e) {
+    e.preventDefault();
+    onSearch(keyword);
+  }
 
-export default function SearchForm() {
   return (
-    <form className="search__form">
+    <form className="search__form" onSubmit={handleSubmit}>
       <h1 className="search__form-title">What's going on in the world?</h1>
       <p className="search__form-text">
         Find the latest news on any topic and save them in your personal
         account.
       </p>
       <div className="search__form-container">
-        <input className="search__form-input" name="keyword"  placeholder="Enter topic" required></input>
-        <button className="search__form-submit" type="submit">Search</button>
+        <input
+          className="search__form-input"
+          name="keyword"
+          placeholder="Enter topic"
+          onChange={handleKeywordChange}
```

Comment:
If you have `onChange` in an input then you need to add attribute `value` to this input.
This is how React controls inputs.
To read more about `controlled components`: https://reactjs.org/docs/forms.html#controlled-components